src/templates/next/components/TableOfContents/TableOfContents.tsx

```diff
@@ -0,0 +1,25 @@
+import { BiRightArrowAlt } from "react-icons/bi"
+import { TableOfContentsProps } from "~/common"
+
+const TableOfContents = ({ headings }: TableOfContentsProps) => {
+  return (
+    <div className="flex flex-col gap-3 pl-5 border-l-4 border-content w-[18.75rem]">
+      <p className="text-lg text-content-strong font-semibold">On this page</p>
+      <div className="flex flex-col gap-3">
+        {headings.map((heading) => (
+          <div className="flex flex-row gap-2 items-start">
+            <BiRightArrowAlt className="size-6 shrink-0 text-interaction-main" />
+            <a
+              href={heading.anchorLink}
```

Comment:
question: this reminds me, it doesn't seem like we are auto-generating the anchor tags for the heading component. Should we do that as well?

This component is likely going to be auto-generated, so users will only say they want a TOC block. The contents will be handled by us.

Reply:
yea, I thought of that but decided not to because I didn't want to couple the anchor link creation with this component

I imagine the anchor link creation should be whenever we create a heading then we add `id=foo` to the heading
Then when the user adds ToC, we get the anchor links from the headings and pass it into this ToC component

Reply:
Yeah oops I mean like creating the anchor tags for the Heading component, so that the anchor links would work here.

Reply:
[2e70a26](https://github.com/isomerpages/isomer-components/pull/34/commits/2e70a26ddf45267e07829c4ba5c376d6b2328bf2)

added `id` to `HeadingProps` for now, not auto populating `id` from `content` as there is no guarantee of headings being unique

I envision it eventually as 
1. user adds Heading, CMS populates `id` for user, whether it's UUID or content with suffix
2. user adds toc, CMS grabs the ids from the Heading (only h1s for now) and generates TOC automatically

Reply:
Yeah I think this solution should be ok (to add the `id` field) but not quite ok with the CMS being the one that adds it. I think eventually it would be the render engine that does this (because it would have context on which headings there are), and we make it completely transparent to the user editing the page.